pkg/cmd/server/server.go

```diff
@@ -78,11 +95,46 @@ func NewCommand() *cobra.Command {
 		},
 	}
 
+	command.Flags().Var(logLevelFlag, "log-level", logLevelFlag.Usage)
 	command.Flags().StringVar(&kubeconfig, "kubeconfig", "", "Path to the kubeconfig file to use to talk to the Kubernetes apiserver. If unset, try the environment variable KUBECONFIG, as well as in-cluster configuration")
 
 	return command
 }
 
+func newLogger(level logrus.Level, hooks ...logrus.Hook) *logrus.Logger {
+	logger := logrus.New()
+	logger.Level = level
+
+	for _, hook := range hooks {
+		logger.Hooks.Add(hook)
+	}
+
+	return logger
+}
+
+func newLogLevelFlag() *flag.Enum {
+	var (
+		sortedLogLevels  = make([]logrus.Level, len(logrus.AllLevels))
+		logLevelsStrings []string
+	)
+
+	// We want to display log levels in increasing order of severity in the flag help.
+	// Create our own sorted copy of the logrus.AllLevels slice (don't want to permute
+	// the library version) (also note that Panic has the lowest value so the compare
+	// in the sort is reversed). Then concat the string representations of these in
+	// the Usage string.
+	copy(sortedLogLevels, logrus.AllLevels)
+
+	sort.Slice(sortedLogLevels, func(i, j int) bool { return logrus.AllLevels[i] > logrus.AllLevels[j] })
+
+	for _, level := range sortedLogLevels {
+		logLevelsStrings = append(logLevelsStrings, level.String())
+	}
+
+	return flag.NewEnum(logrus.InfoLevel.String(), logLevelsStrings...).
+		WithUsage(fmt.Sprintf("the level at which to log (options %s)", strings.Join(logLevelsStrings, " | ")))
```

Comment:
I'd do "the level at which to log. Valid values are %s" followed by `strings.Join(logLevelsStrings, ", ")`

Reply:
I'm tempted to suggest that instead of doing `WithUsage`, just make a helper that can retrieve a `[]string` of sorted log levels, and use that inside of `NewCommand`, like this:
```go
func NewCommand() *cobra.Command {
  sortedLogLevels := getSortedLogLevels()
  logLevelFlag := flag.NewEnum(logrus.InfoLevel.String(), sortedLogLevels...)
  // ...
  command.Flags().Var(logLevelFlag, "log-level", fmt.Sprintf("the level at which to log....."))

Reply:
OK, that's reasonable.